recipes/lora_finetune_distributed.py

```diff
@@ -855,11 +871,22 @@ def train(self) -> None:
                         and self._is_rank_zero
                     ):
                         time_per_step = time.perf_counter() - t0
+                        tokens_per_second = num_tokens / (time_per_step * world_size)
+                        # Calculate MFU and actual FLOPS
+                        mfu_value, actual_flops = mfu.calculate_mfu_and_flops(
+                            model_flops=self._model_flops,
+                            batch_size=len(batch),
+                            step_time=time_per_step,
+                            world_size=world_size,
+                            device=self._device
+                        )
```

Comment:
aside from time_per_step, all other variables seem to be constant. We can probably pass them in the initi and reduce footprint here.